### ISSUE TITLE
clarification: "Location" header bevahior?

### ISSUE THREAD

**Issue by rchincha** [opencontainers/distribution-spec#79]
As per spec, "Location" header is typically of the form:
Location: /v2/<name>/blobs/uploads/<session_id>

Also, 
```
Though the URI format (/v2/<name>/blobs/uploads/<session_id>) for the Location header is specified, clients SHOULD treat it as an opaque url and SHOULD never try to assemble it.
```

However, docker distribution appears to return a fully qualified "Location" header value of the form:
https://<server_ip>:<server_port>/v2/<name>/blobs/uploads/<session_id>

This behavior needs further clarification. What should be a compliant client's expectation from the server? In one case, it needs to prepend the Location header with the (scheme,address,port) and in the other, not.

**Comment by jonjohnsonjr** [comment]
I think we can just defer to [RFC 7231](https://tools.ietf.org/html/rfc7231#section-7.1.2):

> The field value consists of a single URI-reference.  When it has the
   form of a relative reference ([[RFC3986], Section 4.2](https://tools.ietf.org/html/rfc3986#section-4.2)), the final
   value is computed by resolving it against the effective request URI
   ([[RFC3986], Section 5](https://tools.ietf.org/html/rfc3986#section-5)).

**Comment by rchincha** [comment]
So this would mean the client needs to distinguish between the two Location formats, which is fine. Should this be called out in the dist-spec?

**Comment by jzelinskie** [comment]
Absolutely. We should add a link to the RFC in all the tables that document the headers for endpoints -- some of them are even empty for Location.

**Comment by jdolitsky** [comment]
Resolved by #206
